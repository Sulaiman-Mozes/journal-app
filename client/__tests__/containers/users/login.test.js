import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import Login, { LoginPage } from '../../../containers/users/login';

describe('LoginContainer', () => {
  const middlewares = [thunk];
  const mockStore = configureStore(middlewares);
  const initialState = {};
  const store = mockStore(initialState);

  const fakeEvent = {
    target: { name: '', value: '' },
    preventDefault: jest.fn(),
  };

  const props = {
    users: {
      user_data: {
        email: 'email@gmail.com',
        username: 'username',
        token: 'fqgq7f834uegwfiyugh363guyqwfh8wqwfamu8yf134789qufn1463ewqgshuohpve[q93402-9r8971823rqwbf8p9',
      },
      isloggedIn: true,
    },
    loginUser: jest.fn(),
    history: { push: jest.fn() },
  };

  it('should render correctly', () => {
    const LoginContainer = shallow(
      <Provider store={store}>
        <Login {...props} />
      </Provider>,
    );
    expect(LoginContainer).toMatchSnapshot();
  });

  it('should have the following methods', () => {
    const wrapper = shallow(<LoginPage {...props} />);
    expect(wrapper.instance().handleChange(fakeEvent));
    expect(wrapper.instance().handleSubmit(fakeEvent));
  });
});
