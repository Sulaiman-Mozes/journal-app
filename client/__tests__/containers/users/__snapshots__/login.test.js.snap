// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginContainer should render correctly 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeState": Object {},
    }
  }
>
  <Connect(LoginPage)
    history={
      Object {
        "push": [MockFunction],
      }
    }
    loginUser={[MockFunction]}
    users={
      Object {
        "isloggedIn": true,
        "user_data": Object {
          "email": "email@gmail.com",
          "token": "fqgq7f834uegwfiyugh363guyqwfh8wqwfamu8yf134789qufn1463ewqgshuohpve[q93402-9r8971823rqwbf8p9",
          "username": "username",
        },
      }
    }
  />
</ContextProvider>
`;
